lablup/backend.ai-webui
# fix: calculate liveStat by considering all containers

The liveStat of the current session only calculates the stat of the first container. It will miscalculate the CPU usage, etc. So I modified to calculate the liveStat considering all containers. And the calculation method followed the control panel.

ref: https://github.com/lablup/backend.ai-control-panel/blob/main/frontend/src/pages/hub-session-list.js#L842-L928

## How to test
- core: main
- Check the progress of usage column is same as control panel's session utilization.
  - If it's a 2-core allocation and only one core is 100% occupied, the progress should be 100% taken, but if you want to see the CPU usage for the entire session, you're only using 100% of the 200% available. So, it should be **50%** size

---

<Stress test>

1. Install package
   ```Python
   pip install stress
   ```
2. CPU test (compare with `top`)
   ```Python
    stress -c <number of core>
   ```
3. Mem test (compare with `top`)
   ```Python
    stress –m <size>
    # ex) stress -m 2048
   ```
4. GPU test (compare with `watch nvidia-smi`)
   > I tested with `library/colab-python:pytorch2.1` in http://10.100.64.15:8090
   ```Shell
	git clone https://github.com/wilicc/gpu-burn
	cd gpu-burn
	make
	./gpu_burn 60 # will run gpu_burn for 60 seconds
    ```
   

## Updated
For better testing, I updated utilization column UI like control panel.
![image](https://github.com/lablup/backend.ai-webui/assets/28584164/5e195e64-d4e9-4711-b62b-0a78f5b62c8a)



**Checklist:** (if applicable)
